test/e2e/velero_utils.go

```diff
@@ -393,3 +396,74 @@ func waitForVSphereUploadCompletion(ctx context.Context, timeout time.Duration,
 
 	return err
 }
+
+func getVeleroVersion(ctx context.Context, veleroCLI string) (string, error) {
+	cmd := exec.CommandContext(ctx, veleroCLI, "version")
+	fmt.Println("Get Version Command:" + cmd.String())
+	stdout, stderr, err := veleroexec.RunCommand(cmd)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get velero version, stdout=%s, stderr=%s", stdout, stderr)
+	}
+
+	output := strings.Replace(stdout, "\n", " ", -1)
+	fmt.Println("Version:" + output)
+	regCompiler := regexp.MustCompile(`(?i)client\s*:\s*version\s*:\s*(\S+).+server\s*:\s*version\s*:\s*(\S+)`)
+	versionMatches := regCompiler.FindStringSubmatch(output)
+	if len(versionMatches) < 3 {
+		return "", errors.New("Velero version command returned null version")
+	}
+	if versionMatches[1] != versionMatches[2] {
+		return "", errors.New("Velero server and client version are not matched")
+	}
+	return versionMatches[1], nil
+}
+
+func checkVeleroVersion(ctx context.Context, veleroCLI string, expectedVer string) error {
+	tag := expectedVer
+	tagInstalled, err := getVeleroVersion(ctx, veleroCLI)
+	if err != nil {
+		return errors.WithMessagef(err, "Failed to get Velero version")
+	}
+	if strings.Trim(tag, " ") != strings.Trim(tagInstalled, " ") {
+		return errors.New(fmt.Sprintf("Velero version %s is not as expected %s", tagInstalled, tag))
+	}
+	fmt.Printf("Velero version %s is as expected %s\n", tagInstalled, tag)
+	return nil
+}
+
+func installVeleroCLI(version string) (string, error) {
+	name := "velero-" + version + "-" + runtime.GOOS + "-" + runtime.GOARCH
+	postfix := ".tar.gz"
+	tarball := name + postfix
+	cliTarballUrl := "https://github.com/vmware-tanzu/velero/releases/download/" + version + "/" + tarball
+
+	if err := getVeleroCliTarball(tarball, cliTarballUrl); err != nil {
+		return "", errors.WithMessagef(err, "Failed to get Velero CLI tarball")
+	}
+
+	cmd := exec.Command("tar", "-xvf", tarball, "-C", ".")
```

Comment:
Can we extract this into a temporary directory?

Reply:
Yes, it's good point, I will fixed it.